src/modules/hip/kernel/resample.hpp

```diff
@@ -0,0 +1,298 @@
+#include <hip/hip_runtime.h>
+#include "rpp_hip_common.hpp"
+
+// -------------------- Set 0 - resample kernel device helpers  --------------------
+
+__device__ __forceinline__ float resample_hip_compute(float &x, float &scale, float &center, float *lookup, int &lookupSize)
+{
+    float locRaw = x * scale + center;
+    int locFloor = std::floor(locRaw);
+    float weight = locRaw - locFloor;
+    locFloor = std::max(std::min(locFloor, lookupSize - 2), 0);
+    float current = lookup[locFloor];
+    float next = lookup[locFloor + 1];
+    return current + weight * (next - current);
+}
+
+__device__ __forceinline__ void resample_hip_compute(float4 *src_f4, float4 *dst_f4, const float4 *scale_f4, const float4 *center_f4, float *lookup)
+{
+    float4 locRaw_f4 = (*src_f4) * (*scale_f4) + (*center_f4);
+    int4 locFloor_i4 = make_int4(std::floor(locRaw_f4.x), std::floor(locRaw_f4.y), std::floor(locRaw_f4.z), std::floor(locRaw_f4.w));
```

Comment:
can't you use floor(locRaw_f4) instead?

Reply:
Hi Rajy,
locRaw_f4 is float4 variable 
Currently there is no function available in HIP that can do floor operation on float4 variables directly

Reply:
I suggest having a macro for floor(float4) and use hip math instructions instead of std::floor

Reply:
macro for hip created 👍🏼